src/display.h

```diff
@@ -17,7 +17,7 @@ class Display {
     Display(std::vector<std::string> displayNames);
     virtual ~Display();
 
-    void displayImages(std::vector<cv::Mat>& displayImages);
+    void displayImages(std::vector<cv::Mat*>& displayImages);
```

Comment:
If the vector is being passed by reference, I'm not sure the matrices need to be pointers anymore to prevent copying.